### PR TITLE
[BUG] Clustering notebook (and a few docstring typos)

### PR DESCRIPTION
**Pull request by TonyBagnall**
k-medoids is occasionally failing with an empty cluster (see #672). This should be impossible! This PR (hopefully) mitigates against fails on main through making it deterministic by removing the random data resampling. Does not solve it though.

Piggy backed this problem with ongoing docstring and testing changes

### CODE REVIEW

File: aeon/clustering/k_means.py [hunk 1]
```diff
@@ -23,9 +23,8 @@ class TimeSeriesKMeans(TimeSeriesLloyds):
         Method for initializing cluster centers. Any of the following are valid:
         ['kmeans++', 'random', 'forgy'].
     metric : str or Callable, default='dtw'
-        Distance metric to compute similarity between time series. Any of the following
-        are valid: ['dtw', 'euclidean', 'erp', 'edr', 'lcss', 'squared', 'ddtw', 'wdtw',
-        'wddtw'].
+        Distance metric to compute similarity between time series. For valid
+        distance strings, see aeon.distances.SUPPORTED_DISTANCE_FUNCTIONS.
```
Comment (MatthewMiddlehurst) [inline aeon/clustering/k_means.py:27]: While easier to maintain, I don't think this presents the valid options to users very well. Same with `k-medoids`. If there was a link available to the documentation which contained this list that would be fine IMO, but I'm not sure how to achieve that.

Comment (TonyBagnall) [reply aeon/clustering/k_means.py:27]: it was already out of date and we would need some linkage, its just too brittle to duplicate identical lists of strings everywhere distances are used. Perhaps a function call distances.valid_strings() or something?

Comment (TonyBagnall) [reply aeon/clustering/k_means.py]: @chrisholder has done this in the manner used by scikit, which seems a good idea

**Comment by TonyBagnall** [comment]
<p>I've integrated this into the relevant sections</p>

---

 <a href='https://app.reviewnb.com/aeon-toolkit/aeon/pull/673/discussion/#comment-1699533735'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by TonyBagnall** [comment]
<p>I do agree, but its probably something for a future PR</p>

---

 <a href='https://app.reviewnb.com/aeon-toolkit/aeon/pull/673/discussion/#comment-1699533705'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by TonyBagnall** [comment]
<p>there is some explanation of inertia, added a link to sklearns scoring, which is what we use</p>

---

 <a href='https://app.reviewnb.com/aeon-toolkit/aeon/pull/673/discussion/#comment-1699533677'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by TonyBagnall** [comment]
<p>I agree, Ive added a table, but not linked it. The current linkiing (e.g. forecasting notebook) does not translate to the documentation for some reason. </p>

---

 <a href='https://app.reviewnb.com/aeon-toolkit/aeon/pull/673/discussion/#comment-1699533643'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by hadifawaz1999 — APPROVED** [review]
LGTM!

**Pull request merged**